### ISSUE TITLE
Global state in crab_verifier.cpp and crab/array_domain.cpp

### ISSUE THREAD

**Issue by dthaler** [vbpf/ebpf-verifier#113]
src/crab_verifier.cpp has:
```
program_info global_program_info;
```
This is initialized by run_ebpf_analysis() and used by crab\ebpf_domain.hpp.

Similarly, src/crab/array_domain.cpp has:
```
array_map_t global_array_map;
```
which is used by `lookup_array_map()` in array_domain.hpp.
And array_domain.cpp has the following comment associated:
```
/**
    Ugly this needs to be fixed: needed if multiple analyses are
    run so we can clear the array map from one run to another.
**/
```

A better approach would be to not use global state.

Edit: checklist of files include
- [x] src/crab_verifier.cpp
- [x] src/crab/array_domain.cpp
- [x] src/crab/var_factory.cpp
- [x] src/crab_utils/graph_ops.hpp
- [x] src/crab_utils/stats.hpp

**Comment by elazarg** [comment]
I think these are two independent issues (and will be resolved by independent PRs).
Also there's [var_factory.cpp](https://github.com/vbpf/ebpf-verifier/blob/master/src/crab/var_factory.cpp#L19) which uses global mapping from variables to names.

**Comment by dthaler** [comment]
Fixed by #198
